### ISSUE TITLE
pm2 serve <path> -p 8000

### ISSUE THREAD

**Issue by Unitech** [Unitech/pm2#2670]
Serve <path> folder via http node.js server on desired port.
It needs to embed a simple standalone http server with full mime type support

**Comment by Unitech** [comment]
Suggestion:
https://github.com/keymetrics/km-front/tree/development/app/static_server

**Comment by vmarchaud** [comment]
CLI :
`pm2 serve <path> <port>`

Process file :
```
[{
  "name"      : "frontend",
  "script"    : "serve",
  "env": {
    "SERVE_PORT": 8081,
    "SERVE_PATH": "."
  }
}]

```

**Comment by soyuka** [comment]
I'm not found of the `SERVE_PORT` and `SERVE_PATH` variables, they're too generic. Maybe prefixed by `PM2_`? 

Is there no library to handle mimes types? Having the list embed in the source code feels wrong no?

**Comment by vmarchaud** [comment]
I looked into `mime` module that embed another [dependency](https://www.npmjs.com/package/mime-db) that contains a [json list](https://github.com/jshttp/mime-db/blob/master/db.json) and if you look at the list there are a lot of uncommon mime so for me its better to have an hardcoded list of 90% mime used in normal environment than using module for this.
The env variable are indeed generic cause the `Serve.js` could be used outside of pm2 serve, its just a simple http server.

**Comment by soyuka** [comment]
Ok fair enough about the mime types!

**Comment by vmarchaud** [comment]
https://github.com/Unitech/pm2/commit/9ddee41f15966be7ce3587f14ab7dad27f4b3f27

**Comment by soyuka** [comment]
👍

**Comment by Unitech** [comment]
That would be great to add some pmx probes to track:
- 404/min
- mean latency
- hit count / min

**Comment by vmarchaud** [comment]
Published PM2 `2.4.0`
